src/test/java/seedu/address/logic/LogicManagerTest.java

```diff
@@ -92,6 +101,95 @@ public void getFilteredModuleList_modifyList_throwsUnsupportedOperationException
         assertThrows(UnsupportedOperationException.class, () -> logic.getFilteredModuleList().remove(0));
     }
 
+    @Test
+    public void getStalledCommand_validTest() {
+        Command actual = logic.getStalledCommand();
+        assertNull(actual);
+    }
+
+    @Test
+    public void getStalledCommandText_validTest() {
+        String actual = logic.getStalledCommandText();
+        assertNull(actual);
+    }
+
+    @Test
+    public void getGradPad_validTest() {
+        ReadOnlyGradPad actual = logic.getGradPad();
+        Model testModel = new ModelManager();
+        ReadOnlyGradPad expected = testModel.getGradPad();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void getGradPadFilePath_validTest() {
+        Path actual = logic.getGradPadFilePath();
+        Model testModel = new ModelManager();
+        Path expected = testModel.getGradPadFilePath();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void getGuiSettings_validTest() {
+        GuiSettings actual = logic.getGuiSettings();
+        Model testModel = new ModelManager();
+        GuiSettings expected = testModel.getGuiSettings();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void setGuiSettings_validTest() {
+        GuiSettings expected = new GuiSettings();
+        logic.setGuiSettings(expected);
+        GuiSettings actual = logic.getGuiSettings();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void execute_abortCommand_validTest1() throws CommandException, ParseException {
+        model.addModule(CS2103T);
+        logic.execute(CLEAR_COMMAND_WORD);
+        CommandResult expected = new CommandResult(MESSAGE_CONFIRMATION_CANCEL
+                + String.format("\"%s\"", CLEAR_COMMAND_WORD));
+        CommandResult actual = logic.execute("n");
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void execute_abortCommand_validTest2() throws CommandException, ParseException {
+        model.addModule(CS2103T);
+        logic.execute(CLEAR_COMMAND_WORD);
+        CommandResult expected = new CommandResult(MESSAGE_CONFIRMATION_CANCEL
+                + String.format("\"%s\"", CLEAR_COMMAND_WORD));
+        CommandResult actual = logic.execute(LIST_COMMAND_WORD);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void execute_confirmCommand_validTest() throws CommandException, ParseException {
+        model.addModule(CS2103T);
+        logic.execute(CLEAR_COMMAND_WORD);
+        logic.execute(YES_COMMAND_WORD);
+        Command actual = logic.getStalledCommand();
+        assertNull(actual);
```

Comment:
Instead of opening up `Logic` and checking if its `stalledCommand` is null, could we just do `logic.execute(YES_COMMAND_WORD)` a second time instead? If `stalledCommand` was correctly cleared the first time round, then this second execution should return the nothing-to-confirm msg

This way, we don't need the new contracts in the interface xD

Reply:
wow thats genius actl HAHAHA  changed accordingly